fastlane_bot/tests/deterministic/dtest_manager.py

```diff
@@ -0,0 +1,653 @@
+        """
+        Approves and creates test strategies.
+
+        Args:
+            args (argparse.Namespace): The command-line arguments.
+            test_strategies (dict): The test strategies.
+            from_block (int): The block number.
+
+        Returns:
+            dict: All the relevant test strategies
+        """
+        test_strategy_txhashs: Dict[TestStrategy] or Dict = {}
+        for i, (key, arg) in enumerate(test_strategies.items()):
+            arg["w3"] = self.w3
+            test_strategy = TestStrategy(**arg)
+            test_strategy.get_token_approval(
+                args=args, token_id=0, approval_address=self.carbon_controller.address
+            )
+            test_strategy.get_token_approval(
+                args=args, token_id=1, approval_address=self.carbon_controller.address
+            )
+            tx_hash = self.create_strategy(args, test_strategy)
+            test_strategy_txhashs[str(i + 1)] = {"txhash": tx_hash}
+
+        test_strategy_txhashs = self.append_strategy_ids(
+            args, test_strategy_txhashs, from_block
+        )
+        return test_strategy_txhashs
+
+    @staticmethod
+    def overwrite_command_line_args(args: argparse.Namespace) -> TestCommandLineArgs:
+        """
+        Overwrites the command-line arguments with the default main args.
+
+        Args:
+            args (argparse.Namespace): The command-line arguments.
+
+        Returns:
+            TestCommandLineArgs: The default main args.
+        """
+        # Get the default main args
+        default_main_args = TestCommandLineArgs()
+        default_main_args.blockchain = args.network
+        default_main_args.arb_mode = args.arb_mode
+        default_main_args.timeout = args.timeout
+        default_main_args.rpc_url = args.rpc_url
+        return default_main_args
+
+    def get_most_recent_pool_data_path(self, args: argparse.Namespace) -> str:
+        """
+        Gets the path to the most recent pool data file.
+
+        Args:
+            args (argparse.Namespace): The command-line arguments.
+
+        Returns:
+            str: The path to the most recent pool data file.
+        """
+
+        # Use glob to list all directories
+        most_recent_log_folder = [
+            f for f in glob.glob(self.logs_path) if os.path.isdir(f)
+        ][-1]
+        args.logger.debug(f"Accessing log folder {most_recent_log_folder}")
+        return os.path.join(most_recent_log_folder, "latest_pool_data.json")
+
+    def delete_old_logs(self, args: argparse.Namespace):
+        """
+        Deletes all files and directories in the logs directory.
+
+        Args:
+            args (argparse.Namespace): The command-line arguments.
+        """
+        logs = [f for f in glob.glob(self.logs_path) if os.path.isdir(f)]
+        for log in logs:
+            for root, dirs, files in os.walk(log, topdown=False):
+                for name in files:
+                    os.remove(os.path.join(root, name))
```

Comment:
This prevents us from running this if there is currently an instance of the bot running. 

Could we put the logs from dtests in a different folder so that they are not combined with logs from real bot execution?

Also, could we change this to try/except - otherwise I would guess it would prevent multiple instances of dtests running at the same time.